### PR TITLE
Fix tests for accessors device copyable std containers

### PR DESCRIPTION
**Pull request by aleksmesh**
Fix how device copyable device copy std container types are handled in accessor tests

### CODE REVIEW

File: tests/accessor/host_accessor_api_common.h [hunk 1]
```diff
@@ -111,7 +111,8 @@ class run_api_tests {
       auto offset_id =
           util::get_cts_object::id<dims>::get(offset, offset, offset);
       std::remove_const_t<T> data[buff_size];
-      std::iota(data, (data + buff_range.size()), 0);
+      for (int i = 0; i < buff_size; i++)
```
Comment (bader) [inline tests/accessor/host_accessor_api_common.h:114]:
```suggestion
      for (size_t i = 0; i < buff_size; i++)
```

**Review by bader — APPROVED** [review]

File: tests/accessor/host_accessor_api_common.h (hunk 1 above)
Comment (aleksmesh) [reply tests/accessor/host_accessor_api_common.h]: Done

File: tests/common/type_coverage.h [hunk 2]
```diff
@@ -516,8 +516,11 @@ template <template <typename, typename...> class action, typename T,
 void for_device_copyable_std_containers(argsT &&...args) {
   constexpr std::size_t medium_array_size = 5;
   for_all_types<action, actionArgsT...>(
-      type_pack<std::array<T, medium_array_size>, std::optional<T>,
-                std::pair<T, T>, std::tuple<T, T>, std::variant<T>>{},
+      type_pack<
+          std::array<T, medium_array_size>, std::optional<T>, std::pair<T, T>,
+          std::tuple<T, T>,
+          std::variant<T, typename std::conditional<!std::is_same_v<T, int>,
+                                                    int, char>::type>>{},
```
Comment (ProGTX) [inline tests/common/type_coverage.h:523]:
```suggestion
          std::variant<T, std::conditional_t<!std::is_same_v<T, int>,
                                                    int, char>>>{},
```

File: tests/common/value_operations.h [hunk 3]
```diff
@@ -187,24 +187,34 @@ are_equal(const LeftNonArrT& left, const RightNonArrT& right) {
 }
 
 template <typename... Types, typename U>
-bool are_equal(std::tuple<Types...>& left, const U& right) {
+typename std::enable_if_t<!std::is_same_v<std::tuple<Types...>, U>, bool>
```
Comment (ProGTX) [inline tests/common/value_operations.h:190]:
```suggestion
std::enable_if_t<!std::is_same_v<std::tuple<Types...>, U>, bool>
```

File: tests/accessor/host_accessor_api_common.h [hunk 4]
```diff
@@ -111,7 +111,8 @@ class run_api_tests {
       auto offset_id =
           util::get_cts_object::id<dims>::get(offset, offset, offset);
       std::remove_const_t<T> data[buff_size];
-      std::iota(data, (data + buff_range.size()), 0);
+      for (size_t i = 0; i < buff_size; i++)
+        data[i] = value_operations::init<T>(i);
```
Comment (ProGTX) [inline tests/accessor/host_accessor_api_common.h:115]:
```suggestion
      for (size_t i = 0; i < buff_size; i++) {
        data[i] = value_operations::init<T>(i);
      }
```

**Review by ProGTX — APPROVED** [review]

File: tests/accessor/host_accessor_api_common.h (hunk 4 above)
Comment (aleksmesh) [reply tests/accessor/host_accessor_api_common.h]: @ProGTX Done

**Comment by bader** [comment]
@aleksmesh, please, resolve merge conflicts.

**Comment by aleksmesh** [comment]
> @aleksmesh, please, resolve merge conflicts.

Done

**Pull request merged**